code/game/objects/items/devices/lightreplacer.dm

```diff
@@ -77,111 +115,204 @@
 		return
 
 	if(istype(W, /obj/item/stack/sheet/glass/glass))
-		var/obj/item/stack/sheet/glass/glass/G = W
-		if(G.amount - decrement >= 0 && uses < max_uses)
-			var/remaining = max(G.amount - decrement, 0)
-			if(!remaining && !(G.amount - decrement) == 0)
-				to_chat(user, "There isn't enough glass.")
-				return
-			G.amount = remaining
-			if(!G.amount)
-				user.drop_item(G)
-				del(G)
-			AddUses(increment)
-			to_chat(user, "You insert a piece of glass into the [src.name]. You have [uses] lights remaining.")
+		if(!add_glass(CC_PER_SHEET_GLASS, force_fill = 1))
+			user << "<span class='warning'>\The [src] can't hold any more glass!</span>"
 			return
+		var/obj/item/stack/sheet/glass/glass/G = W
+		G.use(1)
+		user << "<span class='notice'>You insert \the [G] into \the [src].</span>"

```

Comment:
Yeah, you can actually see an example of it in the diff at the very top (since this very file had a to_chat example). You can also just do a ctrl+f included files for to_chat for tons of other examples.